src/base/bittorrent/torrentimpl.cpp

```diff
@@ -799,10 +799,21 @@ bool TorrentImpl::needSaveResumeData() const
     return m_nativeStatus.need_save_resume;
 }
 
-void TorrentImpl::saveResumeData(lt::resume_data_flags_t flags)
+void TorrentImpl::requestResumeData(const lt::resume_data_flags_t flags)
 {
     m_nativeHandle.save_resume_data(flags);
-    m_session->handleTorrentSaveResumeDataRequested(this);
+    m_deferredRequestResumeDataInvoked = false;
+
+    m_session->handleTorrentResumeDataRequested(this);
+}
+
+void TorrentImpl::deferredRequestResumeData()
+{
+    if (!m_deferredRequestResumeDataInvoked)
+    {
+        QMetaObject::invokeMethod(this, [this] { requestResumeData(); }, Qt::QueuedConnection);
+        m_deferredRequestResumeDataInvoked = true;
```

Comment:
I don't know what the above line does but... `requestResumeData()` above will always be called after this line right?

Reply:
> I don't know what the above line does but... `requestResumeData()` above will always be called after this line right?

If the key word of your question is "after", then the answer is Yes.